### PR TITLE
Generate and publish axios client

### PR DESCRIPTION
**Pull request by philipstaffordwood**
This PR adds:
* building an axios client for the openapi spec and 
* pushing the built axios client to npm

To prevent failures of pushing the identical version to NPM the circleCI build number is appended to the version

For a sample push (without master filter) look [here](https://circleci.com/gh/GrindrodBank/dokuti/759)

For now this is done on merge to master (there is a separate redmine to do this on tagging)

Implements feature requested in https://github.com/GrindrodBank/dokuti/issues/40

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,48 @@
...
+  generate-axios-client:
+    working_directory: ~/repo
+    docker:
+       - image: openapitools/openapi-generator-cli:latest
+    steps:
+      - checkout
+      - run: mkdir ~/repo/dokuti-typescript-axios
+      - run: |
+              echo "Adding Build $CIRCLE_BUILD_NUM to version"
+              export BUILD=${CIRCLE_BUILD_NUM:-`date  +%Y%M%d%H%M%S`}
```
Comment (VictorSzewczenko) [inline .circleci/config.yml]: Adding this build related meta-data purely avoids publishing conflicting NPM package versions.I do really like the technique though, and will harvest some of this, as I wanted to inject git tags as the version of the open API specification too. However, as someone wanting to pull an axios client for a version of Dokuti (which will soon be versioned by git tag), then seeing long lists of Circle CI build number appended versions of the Axios client on NPM will be a) confusing b) Will result in thousands of versions over time, which may clutter that repo. I recommend avoiding this by using the git tag as the version to inject into the OAS3 spec (which I really like), and build the corresponding version of the axios client for it.

Comment (philipstaffordwood) [reply .circleci/config.yml]: :+1:  will do

File: .circleci/config.yml [hunk 2]
```diff
@@ -84,6 +127,13 @@ workflows:
   version: 2
   build-commit:
     jobs:
+      - generate-axios-client
+      - publish-axios-client:
+          requires:
+            - generate-axios-client
```
[the remark this answers is not in the document]
Comment (philipstaffordwood) [reply .circleci/config.yml]: @VictorSzewczenko there's a separate PR and redmine for this:
 https://github.com/GrindrodBank/dokuti/pull/43

[the remark this answers is not in the document]
Comment (philipstaffordwood) [reply .circleci/config.yml]: I've merged that branch in to simplify things.

File: .circleci/config.yml (hunk 1 above)
Comment (philipstaffordwood) [reply .circleci/config.yml]: done with 046c2f77804a51a51e8b240dc0664e6294fcf6c5

**Comment by philipstaffordwood** [comment]
Updates:
Now always pushes to docker hub and npm on tag.
Pushing an npm tag a second time fails by design.
On merge with master nothing is pushed, but an extra fossa scan is performed.

**Pull request merged**